File: Facade/Dart/VegRestaurant.dart
import 'IRestaurant.dart';
import 'Menu.dart';

class VegRestaurant implements IRestaurant {
  Menu? _menu;

  VegRestaurant() {
    _menu = Menu("Veg Menu");
  }

  @override
  Menu? getMenus() {
    return _menu;
  }
}

File: Facade/Javascript/VegRestaurant.js
import IRestaurant from "./IRestarutant.js";
import Menu from "./Menu.js";

class VegRestaurant extends IRestaurant {
  constructor() {
    super();
    this.menu = new Menu("Veg Menu");
  }

  getMenus() {
    return this.menu;
  }
}

export default VegRestaurant;
